samples/ravendb/simple/Version_6/Server/ShipOrderHandler.cs

```diff
@@ -6,17 +7,10 @@
 
 public class ShipOrderHandler : IHandleMessages<ShipOrder>
 {
-    ISessionProvider sessionProvider;
-
-    public ShipOrderHandler(ISessionProvider sessionProvider)
-    {
-        this.sessionProvider = sessionProvider;
-    }
-
-    public void Handle(ShipOrder message)
+    public async Task Handle(ShipOrder message, IMessageHandlerContext context)
     {
-        var session = sessionProvider.Session;
-        session.Store(new OrderShipped
+        var session = context.GetRavenSession();
```

Comment:
Still need to work out details about how users will access the Raven session from a handler